mlrun/api/db/sqldb/db.py

```diff
@@ -2122,6 +2122,36 @@ def _latest_uid_filter(self, session, query):
             ),
         )
 
+    def _add_artifact_name_and_iter_query(self, query, name=None, iter=None):
+        if not name and not iter:
+            return query
+
+        if name and name.startswith("~"):
+            # Like query
+            iter_prefix = f"{iter}-" if iter else ""
+            return query.filter(Artifact.key.ilike(f"{iter_prefix}%{name[1:]}%"))
+
+        # From here on, it's either exact name match or no name
+        if iter:
+            if name:
+                return query.filter(Artifact.key == f"{iter}-{name}")
+            return query.filter(Artifact.key.ilike(f"{iter}-%"))
+
+        # Exact match, no iter specified
+        # Escape special chars (_,%) since we still need to do a like query because of the iter.
```

Comment:
Don't you need to escape the special chars anyways ?

Reply:
Not really, since I sometimes do a ==, and using escaping in that case would cause equality predicate to fail. I use escaping only for the special cases of doing a like on the name.

Reply:
I did change the like at the beginning of the function to also use the `exact_name`. Note that this actually changes existing (wrong) behavior - up until now we did a like on the name, and this means that a name with underscores would also use these underscores as wildcards. Even better.

Reply:
Yup when I said anyways the other likes were in my head 👌